src/devTools.js

```diff
@@ -90,12 +95,11 @@ export function send(action, state, options, type, instanceId) {
   }, 0);
 }
 
-export function connect(options = {}) {
+export function connect(options = {}, hostnamePromise) {
   const id = generateId(options.instanceId);
-  start(options);
```

Comment:
Wondering – why did we remove this? (Sorry if this was explained earlier and I missed it)

Reply:
I removed it because it called `connectToServer` with no hostname causing a crash. We have two options here: either remove it or make `start` async and obtain the hostname inside. Both of them work and I don't personally mind which one we'll end up with. I reverted it in my last changes, so if you feel like the other approach is better just let me know and I'll do it that way.

Reply:
I think there might be a third option. Why don't we try this:

```js
(async () => {
  await hostnamePromise;
  start(options);
})();
```

Note that this is the same as:

```js
hostnamePromise.then(() => start(options));
```

It's important that you take the time to understand how async functions and promises work. It will help you with this diff, but more importantly it will help you in future diffs.

Reply:
The code you pasted will result in a crash, you have to assign `options.hostname`. But yeah, there's a third way like this:
```
(async () => {
  options.hostname = await hostnamePromise;
  start(options);
})();
```
There are probably even more ways of doing that btw...

Reply:
```js
// either A
const setHostnamePromise = (async () => {
  options.hostname = await hostnamePromise;
})();
// or B
const setHostnamePromise = obtainHostname(options, hostnamePromise);
(async () => {
  await setHostnamePromise;
  start(options);
})();
```